pkg/agent/client.go

```diff
@@ -507,9 +507,10 @@ func (a *Client) remoteToProxy(connID int64, ctx *connContext) {
 			klog.V(2).InfoS("connection EOF", "connectionID", connID)
 			return
 		} else if err != nil {
+			// "use of closed network connection" errors are expected upon receiving CLOSE_REQ
+			// If connID doesn't exist in connManager, we assume the connection was meant to be closed.
 			if _, ok := a.connManager.Get(connID); !ok {
```

Comment:
Seems like we close the connection on the agent side before we actually delete connID from connManager, which makes me think this check is maybe not doing what we want:

https://github.com/kubernetes-sigs/apiserver-network-proxy/blob/6c2477cfa8a6e11752a27bc99b4a06cc9499557c/pkg/agent/client.go#L413-L421

I'm not sure if deleting connID from connManager before ctx.conn.Close() is safe though.